### ISSUE TITLE
Cursor in Disabled Entry

### ISSUE THREAD

**Issue by AlbinoGeek** [fyne-io/fyne#1880]
### Describe the bug:

It is possible to place the cursor within a disable entry, further, it is possible sometimes to duplicate the cursor between both a disabled entry and an enabled entry.

### To Reproduce:

1. Use attached code
2. Select "one-way flight"
3. Click in the first Entry
4. Click in the second Entry a few times (only takes a couple)
5. Click back into the first entry

### Screenshots:
https://user-images.githubusercontent.com/1910461/106277221-93d0a500-61ed-11eb-9a49-1988d561fcb7.mp4
[ 0:17-0:20 and 0:26-0:31 ]

### Example code:

```go
package main

import (
	"time"

	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/data/binding"
	"fyne.io/fyne/v2/widget"
)

var ukDateWithDecimals = "02.01.2006"

func main() {
	a := app.New()
	w := a.NewWindow("Flight Booker")

	d1 := binding.NewString()
	d2 := binding.NewString()

	t1 := widget.NewEntryWithData(d1)
	t2 := widget.NewEntryWithData(d2)

	options := []string{"one-way flight", "return flight"}
	combo := widget.NewSelect(options, func(s string) {
		if s == options[0] {
			t2.Disable()
		} else {
			t2.Enable()
		}
	})

	button := widget.NewButton("Book", func() {})

	w.SetContent(container.NewVBox(combo, t1, t2, button))
	w.ShowAndRun()
}
```

### Device (please complete the following information):
 - **OS:** Fedora Workstation Linux
 - **Version:** 33 (kernel 5.10.10-200)
 - **Go version:** `go1.15.6 linux/amd64`
 - **Fyne version:** `2.0.0`

**Comment by andydotxyz** [comment]
The disabled entry does not have a cursor.
You can select text and copy it out, but you cannot interact with it.

**Comment by AlbinoGeek** [comment]
Am I selecting `""` in the video at those timestamps? (e.g: does selection have a minimum width even if the value is nothing?)

**Comment by andydotxyz** [comment]
Ah I misread your initial report sorry. Yes it does look like it is selecting "" which it clearly should not do.
If you look carefully the width of the item that appears it is narrower than a cursor.

**Comment by AlbinoGeek** [comment]
I can no longer reproduce this on devel.

**Comment by andydotxyz** [comment]
Yes, the fix will be out in v2.0.2 due soon
